File: teckwah_project/server/models/handover_model.py
from sqlalchemy import Column, String, Text, DateTime, Integer, ForeignKey, func, Index, Computed
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.sql import expression

Base = declarative_base()

class HandoverRecord(Base):
    """
    인수인계 레코드 모델
    """
    __tablename__ = "handover"

    handover_id = Column(Integer, primary_key=True, autoincrement=True)
    title = Column(String(255), nullable=False)
    content = Column(Text, nullable=False)
    created_at = Column(DateTime, default=func.now(), nullable=False)
    updated_at = Column(DateTime, onupdate=func.now(), nullable=True)
    # SQL의 STORED GENERATED COLUMN과 매핑
    effective_date = Column(DateTime, Computed("COALESCE(updated_at, created_at)"))
    created_by = Column(String(50), ForeignKey("user.user_id", ondelete="CASCADE"), nullable=False)

    # 날짜별 정렬을 위한 인덱스
    __table_args__ = (
        Index('idx_date_sort', effective_date),
    ) 

File: node-teckwah/server/models/handover.model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Handover = sequelize.define('handover', {
  handover_id: {
    type: DataTypes.STRING(10),
    primaryKey: true,
    allowNull: false
  },
  title: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  content: {
    type: DataTypes.TEXT,
    allowNull: false
  },
  created_by: {
    type: DataTypes.STRING(50),
    allowNull: false,
    references: {
      model: 'user',
      key: 'user_id'
    }
  },
  created_at: {
    type: DataTypes.DATE,
    allowNull: false
  },
  updated_at: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW
  },
  is_notice: {
    type: DataTypes.BOOLEAN,
    defaultValue: false
  }
}, {
  tableName: 'handover',
  timestamps: false,
  indexes: [
    {
      name: 'idx_handover_date',
      fields: ['created_at']
    },
    {
      name: 'idx_handover_notice',
      fields: ['is_notice']
    }
  ]
});

module.exports = Handover;
